codalab/apps/chahub/utils.py

```diff
@@ -0,0 +1,59 @@
+from django.conf import settings
+
+import logging
+import os
+import requests
+import json
+
+logger = logging.getLogger(__name__)
+
+
+def send_to_chahub(endpoint, data):
+    """
+    Does a post request to the specified API endpoint on chahub with the inputted data.
+    :param endpoint: String designating which API endpoint; IE: 'producers/'
+    :param data: Dictionary containing data we are sending away to the endpoint.
+    :return:
+    """
+
+    assert settings.CHAHUB_API_URL
+
+    if not _chahub_online_check():
+        logger.info("Chahub is currently offline. Cancelling.")
+        raise requests.ConnectionError("Chahub is currently offline. Please try again later.")
+
+    url = "{}{}".format(settings.CHAHUB_API_URL, endpoint)
+
+    data = json.dumps(data)
+
+    _chahub_send_data(url, data)
+
+
+def _chahub_online_check():
```

Comment:
Huh, is this new? doesn't this already exist somewhere?

Reply:
Doesn't look like it exists anywhere previously.

Reply:
Looks like it was plucked from:
```python
@task(queue='site-worker')
def do_chahub_retries(limit=None):
    if not settings.CHAHUB_API_URL:
        return

    logger.info("Checking whether ChaHub is online before sending retries")
    try:
        response = requests.get(settings.CHAHUB_API_URL)
        if response.status_code != 200:
            return
    except requests.exceptions.RequestException:
        # This base exception works for HTTP errors, Connection errors, etc.
        return

    logger.info("ChaHub is online, checking for objects needing to be re-sent to ChaHub")```

Maybe useful to make that function point to this so it's DRY and fix the comment?